portfolio-project-matching-app/backend/daoProject.js

```diff
@@ -192,15 +192,15 @@ const getAllProjects = async () => {
     const collectionSnap = await getCollectionSnapshot('projects');
 
     // loop through documents in the snapshot, adding Project objects to array
-    const projects = [];
-    for (const doc of collectionSnap.docs) {
-        const project = Project.fromDocSnapshot(doc.id, doc);
-        // populate owner, users, and technologies fields
-        project.owner = await getOwnerByUserId(project.ownerId);
-        project.users = await getUsersByProjectId(project.id);
-        project.technologies = await getTechnologiesByProjectId(project.id);
-        projects.push(project);
-    }
+    const projects = collectionSnap.docs.map(doc => Project.fromDocSnapshot(doc.id, doc));
+    await Promise.all(projects.map(async (project) => {
+        [project.owner, project.users, project.technologies] = await Promise.all([
+            getOwnerByUserId(project.ownerId),
+            getUsersByProjectId(project.id),
+            getTechnologiesByProjectId(project.id)
+        ]);
+        return project;
+    }));
```

Comment:
Decided to trade out manually looping through projects for the built-in `map` function. Also, leveraged `Promise.all()` wherever possible.

Note: this could actually be improved further. The outer `Promise.all()` causes the inner `Promise.all()` to wait on the previous document to finish. I briefly looked into "flattening" things out s.t. ALL of the documents could run at the same time, but to no avail. Worth looking into.